Add test for cgi module.

moigagoo opened the pull request.

krux02 commented:
I would like to point out that a table lookup is not faster than liner search through ``queryString``. And for the expected length of possilbe query strings I don't think this will bring any improvement. It is just to make people happy who really want to use a Table type.

dom96 commented on tests/stdlib/tcgi.nim:
```diff
@@ -0,0 +1,23 @@
+import cgi, strtabs
+
+suite "Test cgi module":
+  const queryString = "foo=bar&фу=бар&checked=✓&list=1,2,3&with_space=text%20with%20space"
```
This looks suspiciously close to my PR: https://github.com/nim-lang/Nim/pull/7761

dom96 replied:
Could we just get rid of this functionality in favour of what I implemented?

This implementation doesn't support "foo=bar&foo=tes" for example. I'm not 100% on CGI, but if it's anything like URIs then this should be supported.

krux02 replied:
@dom96 your PR has still the issue with the converter.

dom96 replied:
Yes, I'm aware.

dom96 replied:
I would prefer that PR was fixed rather than this was accepted.

moigagoo replied:
Note the the PR that adds the `readData` proc for strings has already been merged, so you should probably remove it in your PR to avoid duplicating functionality.

moigagoo replied:
Also, turns out I'm an idiot and the initial PR was unnecessary to start with. I'm using Jester, so to get the form value I can just use `@"key"` syntax 🤦‍♂️

moigagoo commented:
@krux02 
> I would like to point out that a table lookup is probably slower than liner search through queryString. And for the expected length of possible query strings I don't think this will bring any improvement. It is just to make people happy who really want to use a Table type.

It is linear search under the hood after all, so it is at least that slow. The PR is not about speed, it's about convenience. Currently, to get a certain param value out of a query string, you'd have to iterate over it until you meet the param you need. I'd prefer to just call `[]` on it.

> question, why is the readData proc still in the diff?
I didn't expect Araq to accept the previous PR without tests, so I've added the new commit to the same branch. Then I've submitted the same branch as PR. For some reason, GitHub didn't detect that the commits are already in devel.

I could create a better PR with just tests, but it seems that @dom96 has a better idea.

> Could we just get rid of this functionality in favour of what I implemented?

I have no problem with that. Your PR implements exactly what I need and does so better.

> This looks suspiciously close to my PR: #7761

I didn't know about your PR, sorry.

dom96 commented:
Why merge this? :'(
